CMakeLists.txt

```diff
@@ -102,27 +107,41 @@ function(trusted_firmware_build)
                -DTFM_TOOLCHAIN_PREFIX=${TFM_TOOLCHAIN_PREFIX}
     BUILD_ALWAYS True
     USES_TERMINAL_BUILD True
-    BUILD_BYPRODUCTS ${VENEERS_FILE} ${PSA_API_NS_PATH}
+    BUILD_BYPRODUCTS ${BUILD_BYPRODUCTS}
   )
 
-  # IPC mode source dependencies
-  add_library(tfm_api STATIC IMPORTED)
-  set_target_properties(tfm_api PROPERTIES
+  add_library(tfm_api
+    ${ZEPHYR_TFM_MODULE_DIR}/tf-m-tests/app/os_wrapper_cmsis_rtos_v2.c
+  )
+
+  add_library(interface_lib STATIC IMPORTED)
```

Comment:
`interface_lib` seems to be a too generic name.

Who needs to link to this lib, and why ?
Based on this `PSA_API_NS_PATH` I think the name could be improved.

However, from what I see what you want is probably to ensure that targets depending on `tfm_api` also links to this librarym that is: `target_link_libraries(<target>... tfm_api)` also links with `PSA_API_NS_PATH`.

For this, you should not use imported lib, as that means the build system expects that lib is always present, but in this case that is not true. You are using a generated lib.

So you probably should do something like here:
https://github.com/nrfconnect/sdk-nrfxlib/blob/master/nrf_security/cmake/combine_archives.cmake#L28-L29

Form what I can judged, something like:
```
target_link_libraries(tfm_api INTERFACE ${PSA_API_NS_PATH})
add_dependencies(tfm_api tfm)   # <--- This you already have in line 138
```

Reply:
I separated some of tfm_api into this:

    add_library(tfm_api_helpers
      ${ZEPHYR_TFM_MODULE_DIR}/tf-m-tests/app/os_wrapper_cmsis_rtos_v2.c
      ${ZEPHYR_TFM_MODULE_DIR}/src/zephyr_tfm_log.c
    )

    target_include_directories(tfm_api_helpers
      PRIVATE
      ${ZEPHYR_TFM_MODULE_DIR}/tf-m-tests/CMSIS/RTOS2/Include
      PUBLIC
      ${ZEPHYR_TFM_MODULE_DIR}/trusted-firmware-m/interface/include
    )

    target_link_libraries(tfm_api_helpers PUBLIC
      zephyr_interface
    )

I need tfm_api_helpers to be linked into the generated libs (like `${PSA_API_NS_PATH}`). How do I do this?

Reply:
I've tried to achieve it in many different ways, but I cannot find a way where the generated libs find the symbols from tfm_api_helpers, except to create imported libs like before.

Reply:
Got help offline, done.